test/xbgpu/test_engine.py

```diff
@@ -1452,3 +1465,99 @@ def get_stream_status(stream_name: str) -> bool:
+        _ = self._patch_get_in_item(monkeypatch, capture_stop_chunk_id, client, requests=capture_stop_requests)
+        timestamp_step = n_samples_between_spectra * n_spectra_per_heap
+
+        def heap_factory(batch_index: int, present: np.ndarray) -> list[spead2.send.HeapReference]:
+            timestamp = batch_index * timestamp_step
+            data = np.full((n_channels_per_substream, n_spectra_per_heap, N_POLS, COMPLEX), 10, np.int8)
+            return [
+                spead2.send.HeapReference(gen_heap(timestamp, ant_index, frequency, data))
+                for ant_index in range(n_ants)
+                if present[ant_index]
+            ]
+
+        corrprod_results, beam_results, _, _ = await self._send_data(
+            mock_recv_streams,
+            mock_send_stream,
+            corrprod_outputs,
+            beam_outputs,
+            heap_factory=heap_factory,
+            present=np.ones((n_batches, n_ants), bool),
+            timestamp_step=timestamp_step,
+            n_channels_per_substream=n_channels_per_substream,
+            frequency=frequency,
+            n_spectra_per_heap=n_spectra_per_heap,
+        )
+
+        # NOTE: The results arrays returned are initialised as zeros, and should
+        # therefore still be zero for indices after the `?capture-stop` request
+        # was issued. That is, completely zero data is akin to no data received.
+        for stopped_stream in stopped_stream_names:
+            if stopped_stream in corrprod_results.keys():
+                # NOTE: As per the `DEFAULT_PARAMETERS`, both corrprod_outputs have the same
+                # heap_accumulation_threshold, which is why we can use either one to calculate
+                # the accumulation index where data stopped transmitting.
+                np.testing.assert_equal(
+                    corrprod_results[stopped_stream][
+                        capture_stop_chunk_id * HEAPS_PER_FENGINE_PER_CHUNK // heap_accumulation_threshold[0] :
+                    ],
+                    0,
+                )
+            elif stopped_stream in beam_results.keys():
+                # NOTE: The `beam_results` contain data for each heap (or batch). As a result,
+                # we multiply the chunk ID at which `?capture-stop` was issued by
+                # `HEAPS_PER_FENGINE_PER_CHUNK` to get the specific heap ID at which data
+                # ceased transmitting for a stopped stream.
+                np.testing.assert_equal(
+                    beam_results[stopped_stream][capture_stop_chunk_id * HEAPS_PER_FENGINE_PER_CHUNK :], 0
+                )
+            # Remove this stopped-stream from the master list of data streams
+            all_stream_names.remove(stopped_stream)
+
+        for captured_stream in all_stream_names:
+            if captured_stream in corrprod_results.keys():
+                # NOTE: The `heap_factory` only sends real data. The results buffer has
```

Comment:
Um, it looks to me like it's sending 10+10j. Perhaps you mean that the correlations that result are purely real?

Reply:
Yes, apologies, I'm referring to the results. I noticed that when inspecting the `corrprod_results` ndarray.